### PR TITLE
make Parsers a weakdep

### PR DESCRIPTION
**Pull request by KristofferC**
Rebase of https://github.com/JuliaStrings/InlineStrings.jl/pull/67.

### CODE REVIEW

**Comment by KristofferC** [comment]
A bit ironic that the invalidation CI fails when it reports:

```
Run echo "Invalidations on default branch: 237 (237 via deps)" >> $GITHUB_STEP_SUMMARY
  echo "Invalidations on default branch: [2](https://github.com/JuliaStrings/InlineStrings.jl/actions/runs/9905946948/job/27372841502?pr=78#step:9:2)37 (237 via deps)" >> $GITHUB_STEP_SUMMARY
  echo "This branch: 44 (44 via deps)" >> $GITHUB_STEP_SUMMARY
```

File: Project.toml [hunk 1]
```diff
@@ -6,21 +6,26 @@ version = "1.4.1"
...
 [compat]
 Parsers = "2"
 julia = "1.6"
 
-[weakdeps]
-ArrowTypes = "31f734f8-188a-4ce0-8406-c8a06bd891cd"
-
 [extensions]
 ArrowTypesExt = "ArrowTypes"
+ParsersExt = "Parsers"
```
Comment (ararslan) [inline Project.toml:19]: Question for my own edification, not really related to the PR: If two packages (say InlineStrings and SaxophoneHorse) have extensions for Parsers, both call their extension module `ParsersExt`, and you load InlineStrings, SaxophoneHorse, and Parsers, what happens? That is, does the convention of naming an extension module `<main package><weakdep>Ext` provide functional disambiguation or is it more for the benefit of the human?

File: Project.toml [hunk 2]
```diff
@@ -6,21 +6,26 @@ version = "1.4.1"
...
+
+[extras]
+Arrow = "69666777-d1a9-59fb-9406-91d4454c9d45"
+Parsers = "69de0a69-1ddd-5017-9359-2bf0b02dc9f0"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Serialization = "9e88b42a-f829-5b0c-bbe9-9e923198166b"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+
+
```
Comment (ararslan) [inline Project.toml:28]:
```suggestion
```

**Review by ararslan — APPROVED** [review]

File: Project.toml (hunk 1 above)
Comment (ararslan) [reply Project.toml:19]: Never mind, mostly answered my own question. Extension modules having the same name doesn't cause any issues, so I suppose the naming convention is more for the human's benefit.

Comment (KristofferC) [reply Project.toml:19]: Yes, they have different UUID so the loading system can distinguish them.

**Pull request merged**